### ISSUE TITLE
Internationalization of Project Settings

### ISSUE THREAD

**Issue by AlmirNeeto99** [godotengine/godot#14582]
Hello, Godot Contributors!!!
I'm here to suggest a little Translation at "Project Settings" because sometimes some Brazilians need to change any setting in the application, but has no Translation there.

This will help so many Game Developers that I know here in Brazil.

Thank You.

PS.: I can help the translation if you let me help :)

**Comment by akien-mga** [comment]
That's not possible, project and editor settings are derived from their property name (e.g. `application/config/name`, and there's no way that we could make such code identifiers translated. It would mean that if you change the editor language, your game breaks.

**Comment by YeldhamDev** [comment]
@akien-mga Hum... Before I fixed that [layout switching in different languages bug](https://github.com/godotengine/godot/pull/13298), the cause of said problem was that the names used for saving were tied with the translations, and making that the translations only affected the tabs visually was enough to fix it. Does something similar happens in the Project Settings?

**Comment by akien-mga** [comment]
@YeldhamDev No, but what I understand from this request would be to make it possible to translation settings in the Project Settings.

But currently, settings are derived directly from their non translatable property path/identifier, e.g. `application/config/name` is converted automatically to `Application > Config > Name`.
Translating "Application", "Config", and "Name" would imply changing the whole system so that the identifier is broken down, used to look up translation in a translation table, display those in the GUI while keeping the internal reference to `application/config/name`.

That's doable, but not trivial, and I'm not sure how the translation table would be generated.

All in all I'm not sure if it's worth it, there are plenty of things which are not meant to be translated in Godot, like node names, all properties, etc. You use English identifiers in the code, so if you see them translated in German in the Inspector, you don't know which is which. It's the same for settings, if you want to change the "Anwendung > Konfiguration > Name" German setting from code, you need to find out that it's actually meant to be accessed via "application/config/name".

**Comment by akien-mga** [comment]
TL;DR: The editor interface can be translated, but anything that is used directly in games should not be translated.

**Comment by ZX-WT** [comment]
Anyone can always create a new doc page for the translation of project settings for users to refer to it if translating the project settings directly is not ideal.

**Comment by Calinou** [comment]
See https://github.com/godotengine/godot/issues/11186, which could provide translated descriptions fo Project Settings properties.

**Comment by jacereda** [comment]
I think I have a solution for this, but the translation effort will be considerable. Looking at how big the community is, it might be doable for some languages.

I'll send a draft PR as soon as I think it's in good shape.

<img width="1081" alt="Screenshot 2019-08-26 at 23 38 30" src="https://user-images.githubusercontent.com/4195919/63725520-e66c9b80-c85a-11e9-955c-c7f62d0f99ba.png">

**Comment by txj-mssl** [comment]
When can this problem be solved? Is there an accurate period or version?

**Comment by txj-mssl** [comment]
It's been three years

**Comment by Calinou** [comment]
@txj-mssl In open source software, people work on what they want to work on :slightly_smiling_face: 

There is no ETA for solving this issue. Not to mention [translating project settings could potentially affect the user experience negatively](https://github.com/godotengine/godot/issues/14582#issuecomment-351048089).

**Comment by akien-mga** [comment]
I also still don't think that we should implement this, so closing.

The project settings' name is derived from their identifier like `application/config/name`, which becomes "Application > Config > Name" programmatically. While we *could* do some magic to allow translating the category and final property names, it would break the correspondence between the setting's name in the documentation/code and what you see in the dialog. It would make it actually *harder* to find settings as you would not be able to follow any tutorial/documentation unless it refers to the translated name in your locale.

Note that the class reference can now be translated (only for 4.0 and later, and note that it's going to change significantly in coming months), so project settings descriptions can be localized and the descriptions are shown by the tooltips.

So while the names will stay in English, you can now add a description of what the property controls in your native language. See #37164.

**Comment by txj-mssl** [comment]
I don't know the U3D project settings, but the UE4 project settings can be set to the user's native language.

**Comment by txj-mssl** [comment]
project settings，What I want is to look like Chinese on the outside, but still use English when using code control.

**Comment by akien-mga** [comment]
The system we use for project and editor settings in Godot is much more flexible than the one of most other engines, as we don't need to write UI code for new settings manually. But that comes with some drawbacks like this one.

We have no plans to change the system for the time being. If we do, then internationalization of setting names might be a possibility.

**Comment by txj-mssl** [comment]
I hope can change it, because my English is not very good. Although I know some English words, I can't get information at a glance from the visual appearance. When I see English words, I feel a strong sense of strangeness in addition to the frequently used grammar keywords.

**Comment by txj-mssl** [comment]
akien-mga ，What's your email? I want to send you an email. I sent it to Juan two days ago, but I still haven't replied. I feel very frustrated. I don't know if I should continue to do some simple tutorials on the video website in China.

**Comment by akien-mga** [comment]
> akien-mga ，What's your email? I want to send you an email. 

`remi at godotengine · org`

> I sent it to Juan two days ago, but I still haven't replied. I feel very frustrated.

Please be patient, we get dozens of emails + hundreds of notifications on a daily basis, and the past two days were a weekend where we sometimes try to take time off...

**Comment by txj-mssl** [comment]
It's been 18 days and I still haven't received a reply. I'm disappointed.I have gradually lost interest in Godot.

**Comment by Calinou** [comment]
@txj-mssl Please stop with the demanding attitude, or you may be blocked from Godot repositories.

**Comment by txj-mssl** [comment]
you are right.

**Comment by txj-mssl** [comment]
I'm just complaining, and you're threatening to ban me, so just ban me.

**Comment by akien-mga** [comment]
> It's been 18 days and I still haven't received a reply. I'm disappointed.I have gradually lost interest in Godot.

A reply to what? I didn't get any email from you on May 25 or within following days, as far as I can tell. I just checked my Spam inbox and didn't see any either.

**Comment by txj-mssl** [comment]
My email was to juan, not to you. Never mind, that's it, I'm sorry, I don't want to discuss it any further.

**Comment by txj-mssl** [comment]
Let's pretend this never happened.
